api/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,309 @@
+	AdminNetPolRuleActionDeny AdminNetPolRuleAction = "Deny"
+	// RuleActionAllow enables admins to specifically allow certain traffic.
+	AdminNetPolRuleActionAllow AdminNetPolRuleAction = "Allow"
+)
+
+// AdminNetworkPolicyPeer defines an in-cluster peer to allow traffic to/from.
+// Exactly one of the selector pointers should be set for a given peer. If a
+// consumer observes none of its fields are set, they should assume an unknown
+// option has been specified and fail closed.
+type AdminNetworkPolicyPeer struct {
+	// Namespaces defines a way to select a set of Namespaces
+	// +optional
+	Namespaces *NamespaceSet `json:"namespaces,omitempty"`
+	// NamespacedPods defines a way to select a specific set of pods in
+	// in a set of namespaces
+	// +optional
+	NamespacedPods *NamespaceAndPodSet `json:"namespacedPods,omitempty"`
+}
+
+// NamespaceSet defines a flexible way to select Namespaces in a cluster.
+// Exactly one of the selectors should be set.  If a consumer observes none of
+// its fields are set, they should assume an unknown option has been specified
+// and fail closed.
+type NamespaceSet struct {
+	// Self cannot be "false" when it is set.
+	// If Self is "true" then all pods in the subject's namespace are selected.
+	// +optional
+	Self *bool `json:"self,omitempty"`
+
+	// NotSelf cannot be "false" when it is set.
+	// if NotSelf is "true" then all pods not in the subject's Namespace are selected.
+	// +optional
+	NotSelf *bool `json:"notSelf,omitempty"`
+
+	// NamespaceSelector is a labelSelector used to select Namespaces, This field
+	// follows standard label selector semantics; if present but empty, it selects
+	// all Namespaces.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// SameLabels is used to select a set of Namespaces that share the same values
+	// for a set of labels.
+	// To be selected a Namespace must have all of the labels defined in SameLabels,
+	// and they must all have the same value as the subject of this policy.
+	// If Samelabels is Empty then nothing is selected.
+	// +optional
+	SameLabels []string `json:"sameLabels,omitempty"`
+
+	// NotSameLabels is used to select a set of Namespaces that do not have a set
+	// of label(s). To be selected a Namespace must have none of the labels defined
+	// in NotSameLabels. If NotSameLabels is empty then nothing is selected.
+	// +optional
+	NotSameLabels []string `json:"notSameLabels,omitempty"`
+}
+
+// PodSet defines a flexible way to select pods in a cluster. Exactly one of the
+// selectors should be set.  If a consumer observes none of its fields are set,
+// they should assume an unknown option has been specified and fail closed.
+type PodSet struct {
+	// PodSelector is a labelSelector used to select Pods, This field is NOT optional,
+	// follows standard label selector semantics and if present but empty, it selects
+	// all Pods.
+	PodSelector *metav1.LabelSelector `json:"podselector"`
```

Comment:
```suggestion
	PodSelector *metav1.LabelSelector `json:"podSelector"`
```

also, if non-optional then non-pointer